stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -590,7 +590,19 @@ end
     return C
 end
 
-kron(A::Diagonal{<:Number}, B::Diagonal{<:Number}) = Diagonal(kron(A.diag, B.diag))
+kron(A::Diagonal, B::Diagonal) = Diagonal(kron(A.diag, B.diag))
+
+function kron(A::Diagonal, B::SymTridiagonal)
+    kdv = kron(diag(A), B.dv)
+    kev = _droplast(kron(diag(A), _pushzero(B.ev)))
```

Comment:
This is a bit tricky because `B.ev` for `B::SymTridiagonal` can have as many elements as the main diagonal `B.dv`. So perhaps
```suggestion
    kev = pop!(kron(diag(A), [_evview(B), zero(eltype(B))]))
```

Reply:
Thanks for the pointer. I'll use `_evview`.

Reply:
Done.